teraserver/python/tests/modules/FlaskModule/API/user/test_UserQueryDeviceTypes.py

```diff
@@ -0,0 +1,183 @@
+from tests.modules.FlaskModule.API.BaseAPITest import BaseAPITest
+from libtera.db.models.TeraDeviceType import TeraDeviceType
+
+
+class UserQueryDeviceTypesTest(BaseAPITest):
+    login_endpoint = '/api/user/login'
+    test_endpoint = '/api/user/devicetypes'
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_no_auth(self):
+        response = self._request_with_no_auth()
+        self.assertEqual(response.status_code, 401)
+
+    def test_post_no_auth(self):
+        response = self._post_with_no_auth()
+        self.assertEqual(response.status_code, 401)
+
+    def test_delete_no_auth(self):
+        response = self._delete_with_no_auth(id_to_del=0)
+        self.assertEqual(response.status_code, 401)
+
+    def test_query_no_params_as_admin(self):
+        response = self._request_with_http_auth(username='admin', password='admin')
+        self.assertEqual(response.status_code, 400)
+
+    def test_query_get_as_admin(self):
+        params = {'id_device_type': 0, 'list': False}
+        response = self._request_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.headers['Content-Type'], 'application/json')
+        json_data = response.json()
+
+        params = {'id_device_type': 0, 'list': True}
+        response = self._request_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.headers['Content-Type'], 'application/json')
+        json_data = response.json()
+        self.assertEqual(len(json_data), 4)
```

Comment:
Returned values should be check to ensure they contain all the fields that the API is supposed to return. Something like that:
        for data_item in json_data:
            self._checkJson(json_data=data_item)

Where:
    def _checkJson(self, json_data, minimal=False):
        self.assertGreater(len(json_data), 0)
        self.assertTrue(json_data.__contains__('id_device_type'))

Also adds other fields that are returned, key and name.